test/extended/machines/scale.go

```diff
@@ -186,6 +187,35 @@ func scaleMachineSet(name string, replicas int) error {
 }
 
 var _ = g.Describe("[sig-cluster-lifecycle][Feature:Machines][Serial] Managed cluster should", func() {
+
+	var (
+		c      *kubernetes.Clientset
+		dc     dynamic.Interface
+		helper *bmhelper.BaremetalTestHelper
+	)
+
+	g.BeforeEach(func() {
+
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		c, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// For baremetal platforms, an extra worker must be previously
+		// deployed to allow subsequent scaling operations
+		helper = bmhelper.NewBaremetalTestHelper(dc)
+		if helper.CanDeployExtraWorkers() {
+			helper.Setup()
+			helper.DeployExtraWorker(0)
+		}
+	})
+
+	g.AfterEach(func() {
+		helper.DeleteAllExtraWorkers()
+	})
+
```

Comment:
When scaling down the number of replicas, would it be possible to define that a particular extra worker should be scaled down using the equivalent of "oc annotate machine <extra-worker> machine.openshift.io/cluster-api-delete-machine=yes"? Perhaps the test is already doing that and I missed it.

Reply:
Yes @bfournie the test is exactly using that approaching, by marking the newly created machine with the `cluster-api-delete-machine` annotation (see [here](https://github.com/openshift/origin/blob/838021cc5dfa1a75b6efacfc77ecae65850f564e/test/extended/machines/scale.go#L260-L262))